keystorage/aws/src/main/java/tech/pegasys/signers/aws/AwsSecretsManager.java

```diff
@@ -68,6 +80,39 @@ public Optional<String> fetchSecret(final String secretName) {
     }
   }
 
+  private Collection<SecretListEntry> listSecrets(final AbstractMap<String, String> tags) {
+    final ListSecretsRequest.Builder listSecretsRequestBuilder = ListSecretsRequest.builder();
+    if (tags != null) {
+      final Filter keys = Filter.builder().key("tag-key").values(tags.keySet()).build();
+      final Filter values = Filter.builder().key("tag-value").values(tags.values()).build();
+      listSecretsRequestBuilder.filters(keys, values);
+    }
+    final ListSecretsRequest listSecretsRequest = listSecretsRequestBuilder.build();
+    return secretsManagerClient.listSecrets(listSecretsRequest).secretList();
```

Comment:
How does this compare performance wise with using the listSecretVersionIdsPaginator approach? Looks like this will force all the retrieval of the secrets metadata upfront before retrieval the secret values themselves.